xla/primitive_util.h

```diff
@@ -408,15 +426,20 @@ inline bool CastPreservesValues(PrimitiveType from_type,
   if (primitive_util::IsComplexType(from_type)) {
     return false;
   }
-  // F -> F is safe if the exponent and significand are preserved.
+  // F -> F is safe if the exponent, significand are preserved, and the
```

Comment:
To make more grammatical, replace "exponent, significand" with "exponent/significand",  remove the comma after "preserved", and remove the word "the" before "to_type".

```
// F -> F is safe if the exponent/significand are preserved and `to_type`
// preserves infinities in `from_type.
```